backend/sirius-web-spring-collaborative-diagrams/src/main/java/org/eclipse/sirius/web/spring/collaborative/diagrams/handlers/InvokeDeleteToolOnDiagramEventHandler.java

```diff
@@ -0,0 +1,136 @@
+        this.objectService = Objects.requireNonNull(objectService);
+        this.diagramQueryService = Objects.requireNonNull(diagramQueryService);
+        this.toolService = Objects.requireNonNull(toolService);
+        this.messageService = Objects.requireNonNull(messageService);
+
+        // @formatter:off
+        this.counter = Counter.builder(Monitoring.EVENT_HANDLER)
+                .tag(Monitoring.NAME, this.getClass().getSimpleName())
+                .register(meterRegistry);
+        // @formatter:on
+    }
+
+    @Override
+    public boolean canHandle(IDiagramInput diagramInput) {
+        return diagramInput instanceof InvokeDeleteToolOnDiagramInput;
+    }
+
+    @Override
+    public void handle(One<IPayload> payloadSink, Many<ChangeDescription> changeDescriptionSink, IEditingContext editingContext, IDiagramContext diagramContext, IDiagramInput diagramInput) {
+        this.counter.increment();
+
+        String message = this.messageService.invalidInput(diagramInput.getClass().getSimpleName(), InvokeNodeToolOnDiagramInput.class.getSimpleName());
+        IPayload payload = new ErrorPayload(diagramInput.getId(), message);
+        ChangeDescription changeDescription = new ChangeDescription(ChangeKind.NOTHING, diagramInput.getRepresentationId(), diagramInput);
+
+        if (diagramInput instanceof InvokeDeleteToolOnDiagramInput) {
+            InvokeDeleteToolOnDiagramInput input = (InvokeDeleteToolOnDiagramInput) diagramInput;
+            Diagram diagram = diagramContext.getDiagram();
+            // @formatter:off
+            var optionalTool = this.toolService.findToolById(editingContext, diagram, input.getToolId())
+                    .filter(DeleteTool.class::isInstance)
+                    .map(DeleteTool.class::cast);
+            // @formatter:on
+            if (optionalTool.isPresent()) {
+                IStatus status = this.executeTool(editingContext, diagramContext, input.getDiagramElementId(), optionalTool.get());
+                if (status instanceof Success) {
+
+                    payload = new InvokeDeleteToolOnDiagramSuccessPayload(diagramInput.getId(), diagram);
+                    changeDescription = new ChangeDescription(ChangeKind.SEMANTIC_CHANGE, diagramInput.getRepresentationId(), diagramInput);
+                }
+            }
+        }
+
+        payloadSink.tryEmitValue(payload);
+        changeDescriptionSink.tryEmitNext(changeDescription);
+    }
+
+    private IStatus executeTool(IEditingContext editingContext, IDiagramContext diagramContext, UUID diagramElementId, DeleteTool tool) {
+        IStatus result = new Failure(""); //$NON-NLS-1$
+        Diagram diagram = diagramContext.getDiagram();
+        Optional<Node> node = this.diagramQueryService.findNodeById(diagram, diagramElementId);
+        Optional<Object> self = Optional.empty();
+        if (node.isPresent()) {
+            self = this.objectService.getObject(editingContext, node.get().getTargetObjectId());
+        }
```

Comment:
We could consider adding the selected edge if that what the tool is trying to delete...